browser/resources/settings/brave_leo_assistant_page/brave_leo_assistant_page.ts

```diff
@@ -113,18 +130,22 @@ class BraveLeoAssistantPageElement extends BraveLeoAssistantPageBase {
     }
 
     computeDisplayName_() {
-      const model = this.models_?.find(
-        (model) => model.model.key === this.defaultModelKeyPrefValue_
+      const foundEntry = this.models_?.find(
+        (entry) => {
+          return entry.model.key === this.defaultModelKeyPrefValue_
+        }
       )
-      if (!model) {
+
+      if (!foundEntry) {
+        console.error('Cant compute display name of the model because model not found in models list')
         return '' // It should appear as if nothing is selected
       }
-      return model.model.displayName
+
+      return foundEntry.model.displayName
     }
 
     onModelSelectionChange_(e: any) {
-      this.setPrefValue(MODEL_PREF_PATH, e.value)
-      this.defaultModelKeyPrefValue_ = e.value
+      this.browserProxy_.getSettingsHelper().changeDefaultModel(e.value)
```

Comment:
yea, but there's a strict check now to ensure if the changed model is part of the list or not and we report to an observer event that default model key has changed.

Reply:
actually, I have removed the does_model_exist check because some unit tests wanted to set keys to obsolete ones for evaluation, and this logic was rejecting them, causing the tests to fail. I left it flexible with no checks and only an observer event.

Reply:
I think my review comment was a mistake because I knew about that later on in the review and suggested removing the pref from being public entirely. But now if you've removed the main reason for the function to exist then you could go back to settings code changing the default model via prefs, directly. Notifying observers doesn't seem like a reason for owning the preference because the service can just use a pref observer. If the "does_model_exist" check _was_ important though, then you could modify the test.

Reply:
this is reverted back to the logic with strict checks. tests can use `SetDefaultModelKeyWithoutValidationForTesting`